src/modules/commander/accelerometer_calibration.cpp

```diff
@@ -465,36 +465,64 @@ calibrate_return do_accel_calibration_measurements(orb_advert_t *mavlink_log_pub
 	uint64_t timestamps[max_accel_sens];
 
 	// We should not try to subscribe if the topic doesn't actually exist and can be counted.
-	const unsigned accel_count = orb_group_count(ORB_ID(sensor_accel));
-	for (unsigned i = 0; i < accel_count; i++) {
-		worker_data.subs[i] = orb_subscribe_multi(ORB_ID(sensor_accel), i);
-		if (worker_data.subs[i] < 0) {
+	const unsigned orb_accel_count = orb_group_count(ORB_ID(sensor_accel));
+
+	// Warn that we will not calibrate more than max_accels accelerometers
+	if (orb_accel_count > max_accel_sens) {
+		calibration_log_critical(mavlink_log_pub, "[cal] Detected %u accels, but will calibrate only %u", orb_accel_count, max_accel_sens);
+	}
+
+	for (unsigned cur_accel = 0; cur_accel < orb_accel_count && cur_accel < max_accel_sens; cur_accel++) {
+
+		// Lock in to correct ORB instance
+		bool found_cur_accel = false;
+		for(unsigned i = 0; i < orb_accel_count && !found_cur_accel; i++) {
+			worker_data.subs[cur_accel] = orb_subscribe_multi(ORB_ID(sensor_accel), i);
+
+			struct accel_report report = {};
+			orb_copy(ORB_ID(sensor_accel), worker_data.subs[cur_accel], &report);
+
+#if !defined(__PX4_QURT) && !defined(__PX4_POSIX_RPI) && !defined(__PX4_POSIX_BEBOP)
+
+			// For NuttX, we get the UNIQUE device ID from the sensor driver via an IOCTL
+			// and match it up with the one from the uORB subscription, because the
+			// instance ordering of uORB and the order of the FDs may not be the same.
+
+			if(report.device_id == device_id[cur_accel]) {
+				// Device IDs match, correct ORB instance for this accel
+				found_cur_accel = true;
+				// store initial timestamp - used to infer which sensors are active
+				timestamps[cur_accel] = report.timestamp;
+			} else {
+				orb_unsubscribe(worker_data.subs[cur_accel]);
+			}
+
+#elif defined(__PX4_QURT) || defined(__PX4_POSIX_RPI) || defined(__PX4_POSIX_BEBOP)
```

Comment:
Done. Also extended to include Excelsior boards. 

Side note : These defines are yuck.